### ISSUE TITLE
test: add more edge case tests for existing rules

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#301]
## Good First Issue

This is a testing-focused issue suitable for newcomers who want to learn the codebase.

## Description

Several rules could benefit from additional edge case tests. This is a great way to learn how rules work while improving test coverage.

## Suggested Areas

Pick any rule and add tests for edge cases like:
- Empty files
- Files with only whitespace
- Unicode characters in headings/content
- Very long lines
- Nested structures (lists in blockquotes, etc.)
- Mixed content types

## How to Find Rules Needing Tests

```bash
# Run tests and look at coverage
cargo test --lib

# Look at existing test patterns
grep -r "#\[test\]" crates/mdbook-lint-rulesets/src/
```

## Example Test Pattern

```rust
#[test]
fn test_md001_empty_file() {
    let content = "";
    let document = Document::new(content.to_string(), PathBuf::from("test.md")).unwrap();
    let rule = MD001::default();
    let violations = rule.check(&document).unwrap();
    assert!(violations.is_empty());
}

#[test]
fn test_md001_unicode_headings() {
    let content = "# Hello World\n## Subsection";
    let document = Document::new(content.to_string(), PathBuf::from("test.md")).unwrap();
    let rule = MD001::default();
    let violations = rule.check(&document).unwrap();
    assert!(violations.is_empty());
}
```

## Getting Started

1. Fork and clone the repository
2. Pick a rule in `crates/mdbook-lint-rulesets/src/standard/` or `crates/mdbook-lint-rulesets/src/mdbook/`
3. Add edge case tests
4. Run `cargo test --lib` to verify
5. Submit a PR

## Resources

- [Contributing guide](https://joshrotenberg.github.io/mdbook-lint/contributing.html)
- Look at existing tests in rule files for patterns

**Comment by joshrotenberg** [comment]
Triage decision: **needs clarification and a finite scope**.

As written, this is an evergreen testing suggestion rather than a task with a completion condition. Before implementation, it should identify a concrete checklist of rules and uncovered branches—ideally from the current coverage report—and define when the issue is complete.

A useful scope would be a small batch such as 3–5 named rules, with specific missing cases for each. Additional batches can remain good-first-issue follow-ups instead of keeping one permanently open-ended issue.
